python/ray/autoscaler/local/node_provider.py

```diff
@@ -90,12 +111,75 @@ class LocalNodeProvider(NodeProvider):
     """
 
     def __init__(self, provider_config, cluster_name):
+        self.auto_scheduling_mode = False
+        if "server_address" in provider_config:
+            self.auto_scheduling_mode = True
+            self.server_address = provider_config["server_address"]
+            provider_config, cluster_name = self._get_node_configs(
+                provider_config, cluster_name)
+
         NodeProvider.__init__(self, provider_config, cluster_name)
-        self.state = ClusterState("/tmp/cluster-{}.lock".format(cluster_name),
-                                  "/tmp/cluster-{}.state".format(cluster_name),
-                                  provider_config)
+        self.state = ClusterState(
+            "/tmp/cluster-{}.lock".format(cluster_name),
+            "/tmp/cluster-{}.state".format(cluster_name),
+            provider_config,
+        )
+
+    def _get_node_configs(self, provider_config, cluster_name):
+        # Concatenate user_id to make cluster names unique between users.
+        new_cluster_name = cluster_name + "_" + getpass.getuser()
+        request = {
+            "request_type": "get_node_ips",
+            "cluster_name": new_cluster_name,
+            "num_workers": provider_config["max_workers"],
+        }
+        node_ips = self.get_http_response(request, self.server_address)
+        if "head_ip" not in node_ips:
+            raise OSError("Not enough nodes are available." +
```

Comment:
This check doesn't seem right, why is it raised if head_ip is not found? Also, OSError is not an appropriate type.